### PR TITLE
geometry: a skeleton for the MS-RDPEGT channel

### PR DESCRIPTION
**Pull request by hardening**
The base for implementing the MS-RDPEGT client channel that allows to track window geometry.

### CODE REVIEW

File: channels/geometry/client/geometry_main.c [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+	Stream_Read_INT32(s, rgndata->boundingRect.x);
+	Stream_Read_INT32(s, rgndata->boundingRect.y);
+	Stream_Read_INT32(s, right);
+	Stream_Read_INT32(s, bottom);
+
+	rgndata->boundingRect.width = right - rgndata->boundingRect.x;
+	rgndata->boundingRect.height = bottom - rgndata->boundingRect.y;
+
+	len -= 32;
+	if (len / 4 < rgndata->nRectCount)
```
Comment (mfleisz) [inline channels/geometry/client/geometry_main.c]: I think this should be ```len / 16``` (4 * sizeof(UINT32))

File: include/freerdp/channels/geometry.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+enum {
+	GEOMETRY_UPDATE	= 1,
+	GEOMETRY_CLEAR	= 2
+};
+
+enum {
+	RDH_RECTANGLE = 1
+};
+
+struct _FREERDP_RGNDATA {
```
Comment (mfleisz) [inline include/freerdp/channels/geometry.h]: ```{``` Formatting (same for the enums above)

File: channels/geometry/client/geometry_main.c (hunk 1 above)
Comment (hardening) [reply channels/geometry/client/geometry_main.c]: good catch

**Comment by hardening** [comment]
@mfleisz 's remarks have been addressed

**Comment by hardening** [comment]
@mfleisz are the changes ok for you ?

**Pull request merged**